src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -36,6 +36,11 @@
     # times if the source is the same as well.
     classpath = dict()
     for dep in report['dependencies']:
+        if not dep.get('file'):
+            raise Exception(
```

Comment:
I'm not certain this error message is right -- in the case of #14577, we're getting a null file field because of an incorrect POM in `repo1.maven.org`, but there is also a repo that provides the JAR. So this message doesn't direct the user to a meaningful fix, and I think we're going to need to patch Coursier before fixing the repo list actually solves this problem.

At the moment, the best way to deal with this is to hand-configure a `jvm_artifact` for the problematic dependency, using the `jar=` field or the `url=` field.

Note also that this error _should_ only come up for transitive dependencies; coursier usually hard-fails for direct dependencies that are misconfigured, and the error message out of coursier is actually useful (it shows which repos were tested).

Reply:
So given the pending cache fixes, this message will soon be correct, so should be OK to merge?

Reply:
Yes, once that's done that will be an excellent first step in debugging. We may want to point out the `url=` and `jar=` options as well (in case adding more maven repos doesn't help).